internal/upgradecheck/http_test.go

```diff
@@ -67,6 +68,14 @@ func TestCheckForUpgrades(t *testing.T) {
 	ctx := logging.NewContext(context.Background(), logging.Discard())
 	cfg := &rest.Config{}
 
+	checkData := func(header string) {
+		data := clientUpgradeData{}
+		err := json.Unmarshal([]byte(header), &data)
+		assert.NilError(t, err)
```

Comment:
🔧 `checkData` is called from within subtests, so it should take a `*testing.T` for the asserts here.

🤔 I forget exactly what happens when asserting on the wrong `T`, but it makes it hard to diagnose the error. (It panics maybe?)

Reply:
Good note, I've updated the test accordingly and put a link to [the go docs](https://pkg.go.dev/github.com/stretchr/testify/assert#hdr-Assertions) that talk a bit about why `T` is passed.

Reply:
`assert` here is [gotest.tools/v3/assert](https://pkg.go.dev/gotest.tools/v3/assert) so we should probably link to that, if it explains.

When it's the wrong `T`, it prints `test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test` with none of the messages from the assert package.

https://go.dev/play/p/gIVKDrd9rTR

Reply:
:thinking: Looking through that link and the code a bit, I can't find anything that exactly points that out. I'll update the note and remove the link.